### ISSUE TITLE
Question About Training Loss

### ISSUE THREAD

**Issue by Cedric-Perauer** [PRBonn/segcontrast#20]
Dear authors, 

thank you for publishing this great research (as well as the follow-up work TARL). We run into some issues when running the contrastive pre-training (which we would eventually like to run on our own data) on semantic-kitti. 

We have tried it on different machines, one with a newer GPU where we use Cuda 11.8 as well as another one where we use Cuda 11.1 with ME 0.5.4, torch 1.9 and torchvision 0.10 as you described in this issue https://github.com/PRBonn/segcontrast/issues/11 and the results are consistent. 
I have seen the issue https://github.com/PRBonn/segcontrast/issues/18 where you kindly provided the training loss curves already. 
We are having some issues with the loss as our curves look like this : 

![photo_2024-01-19 1 10 01 PM](https://github.com/PRBonn/segcontrast/assets/40869163/85059d18-a7be-4409-b71b-85ca67de843d)

<img width="449" alt="Screenshot 2024-01-19 at 1 10 41 PM" src="https://github.com/PRBonn/segcontrast/assets/40869163/d2dbe01d-394a-4e56-b334-9cf0f7a1335e">

Do you know what could be the issue ? 

Thank you for your help !

**Comment by nuneslu** [comment]
I'm not sure what can be the problem. Could you share the log file here? Also, could you share the command you used to run it and the config parameters?

**Comment by Cedric-Perauer** [comment]
Thanks for your fast reply, before the progress bar appears we get this log : 

```bash
The size of train data is 19130
The size of validation data is 4071
GPU available: True, used: True
TPU available: None, using: 0 TPU cores
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]

  | Name       | Type             | Params
------------------------------------------------
0 | moco_model | MoCo             | 43.5 M
1 | criterion  | CrossEntropyLoss | 0     
------------------------------------------------
21.7 M    Trainable params
21.7 M    Non-trainable params
43.5 M    Total params
``` 
As for the command we use the standard one from README.md (standard parameters from contrastive_train.py file are used)

```
python3 contrastive_train.py --use-cuda --use-intensity --segment-contrast --checkpoint segcontrast
``` 
In the meantime we also tried running TARL (on same machine) which works great.

**Comment by nuneslu** [comment]
Could you also provide the `events.out` log files from tensorboard?

**Comment by Cedric-Perauer** [comment]
Thank you for that question !! I actually kept the training running for longer now to provide good logs and the loss does decrease for me now after about ~900 iterations (so a bit longer than in the curves you provided). In our previous runs we only ran to 500-600 iterations (iteration count seemed to be higher in our wandb logging), so sorry to bother about this issue and thanks for the help ! 

I could do a full training run over the weekend with the current setup (on this machine we use Ada GPU and Cuda 11.8 with ME 0.5.4 and torch 2.0.1) and also test the standard supervised training in case it helps for someone with newer GPUs wants to use the codebase in the future ?

**Comment by Cedric-Perauer** [comment]
Thank you for that question !! I actually kept the training running for longer now to provide good logs and the loss does decrease for me now after about ~900 iterations (so a bit longer than in the curves you provided). In our previous runs we only ran to 500-600 iterations (iteration count seemed to be higher in our wandb logging), so sorry to bother about this issue and thanks for the help ! After about ~5k iterations loss look similar to the curves you provided. 

Just one more question : how long is a single epoch supposed to take after the first epoch ? 
Is it correct to assume that for single GPU setup TARL is much faster to train (looks like epoch count is the same from the papers) ?

**Comment by nuneslu** [comment]
Hi! Sorry for the late reply. The first epoch should take longer due to the pre-processing of the segments. After that, it should take around 3~4h each epoch. Overall they should have similar training time I guess.
